### PR TITLE
Request to merge milestone 3 (game logic feature) to development branch

### PR DESCRIPTION
**Pull request by VanessaAoki**
This is the branch for Milestone 3 (game logic) in the Tic Tac Toe ruby project.

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      puts "That's an invalid move, try again: "
+      turn
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.any? do |combo|
+      if position_taken?(combo[0]) && @board[combo[0]] == @board[combo[1]] && @board[combo[1]] == @board[combo[2]]
+        return combo
+      end
```
Comment (ElijahTobs) [inline lib/game.rb:61]: Ruby methods automatically returns the last evaluated statement therefore, there is no need for explicit declaration `return`

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @VanessaAoki and @DsunilK :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] No `puts/print` and `gets.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] Your commit messages should be capitalized, and written in the imperative tense, not past tense. You can watch this [video](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6) on how to edit your commit messages using git rebase.
* [x] Make sure your game doesn't take accept blank spaces as player names
image
* [x] While playing your game, it was difficult to know whose turn it was as your game didn't display the players' name when it was their turn.
* [x] Here, player X won but the board wasn't displayed immediately after the player input. The board should always be displayed immediately after a player makes a valid move. So, in this case, the board should be displayed before the congratulatory message.
image

##### Implementation details

* [x] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))


## NICE TO HAVE
* [x] An invalid move isn't same as a taken position. It will be nice if you can make that differentiation.
* [x] See inline comment below.



### Make the requested changes and submit for another review.
## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

File: lib/game.rb (hunk 1 above)
Comment (SunilDevalla) [reply lib/game.rb:61]: Thank you for the feedback, we will modify the changes suggested and request another review.

**Review by Georjane — APPROVED** [review]
## Status: PR Approved ✔️ ✔️
### Hi Team 🙌
Congratulations on successfully completing this milestone. Nice work done. You can go ahead and merge this PR. 

Good luck on the next milestones and Happy Coding 💻

**Pull request merged**